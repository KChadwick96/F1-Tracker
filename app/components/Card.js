import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const Card = ({children, title, actionText, onActionPress}) => {
    const transformedTitle = title.toUpperCase();

    return (
        <View style={styles.container} elevation={1}>
            <Text style={styles.title}>{transformedTitle}</Text>
            {children}
        </View>
    )
}

Card.propTypes = {
    children: PropTypes.any.isRequired,
    title: PropTypes.string.isRequired,
    actionText: PropTypes.string,
    onActionPress: PropTypes.func
};

const styles = StyleSheet.create({
    container: {
        padding: 15,
        marginHorizontal: 10,
        borderRadius: 4,
        backgroundColor: 'white',
        shadowColor: '#7a7a7a',
        shadowRadius: 3,
        shadowOpacity: 0.5
    },
    title: {
        color: 'black',
        marginBottom: 15,
        fontWeight: 'bold',
        color: 'grey',
        fontSize: 15
    }
});

export default Card;