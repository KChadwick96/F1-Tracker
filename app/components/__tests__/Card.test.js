import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';

import Card from '../Card';

const component = <Card title="Up Next">
    <Text>Hello</Text>
</Card>;

it('renders correctly', () => {
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
});