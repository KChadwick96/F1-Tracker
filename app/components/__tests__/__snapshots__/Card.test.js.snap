// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  elevation={1}
  style={
    Object {
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Up Next
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello
  </Text>
</Component>
`;
